spacesvm/tests/vm/client.rs

```diff
@@ -0,0 +1,242 @@
+        };
+
+        // in this context we don't care about the response unless its an error
+        let resp = self.inner.initialize(request).await.map_err(|e| {
+            Error::new(
+                ErrorKind::Other,
+                format!("initialize request failed: {:?}", e),
+            )
+        });
+        assert!(resp.is_ok());
+        Ok(())
+    }
+
+    async fn set_state(&self, _state: avalanche_types::subnet::rpc::snow::State) -> Result<()> {
+        // TODO:
+        Ok(())
+    }
+
+    async fn shutdown(&self) -> Result<()> {
+        // TODO:
+        Ok(())
+    }
+
+    async fn version(&self) -> Result<String> {
+        Ok(String::new())
+    }
+
+    async fn create_static_handlers(
+        &mut self,
+    ) -> Result<HashMap<String, avalanche_types::subnet::rpc::common::http_handler::HttpHandler>>
+    {
+        let resp = self
+            .inner
+            .create_static_handlers(avalanche_types::proto::google::protobuf::Empty {})
+            .await
+            .map_err(|e| {
+                Error::new(
+                    ErrorKind::Other,
+                    format!("create static handler request failed: {:?}", e),
+                )
+            })?;
+
+        let resp = resp.into_inner();
+
+        let mut http_handler: HashMap<
+            String,
+            avalanche_types::subnet::rpc::common::http_handler::HttpHandler,
+        > = HashMap::new();
+
+        for h in resp.handlers.iter() {
+            let lock_option =
+                avalanche_types::subnet::rpc::common::http_handler::LockOptions::try_from(
+                    h.lock_options,
+                )
+                .map_err(|_| Error::new(ErrorKind::Other, "invalid lock option"))?;
+            http_handler.insert(
+                h.prefix.clone(),
+                avalanche_types::subnet::rpc::common::http_handler::HttpHandler {
+                    lock_option,
+                    handler: None,
+                    server_addr: Some(h.server_addr.clone()),
+                },
+            );
+        }
+
+        Ok(http_handler)
+    }
+
+    async fn create_handlers(
+        &mut self,
+    ) -> Result<HashMap<String, avalanche_types::subnet::rpc::common::http_handler::HttpHandler>>
+    {
+        let resp = self
+            .inner
+            .create_handlers(avalanche_types::proto::google::protobuf::Empty {})
+            .await
+            .map_err(|e| {
+                Error::new(
+                    ErrorKind::Other,
+                    format!("create handler request failed: {:?}", e),
+                )
+            })?;
+
+        let resp = resp.into_inner();
+
+        let mut http_handler: HashMap<
+            String,
+            avalanche_types::subnet::rpc::common::http_handler::HttpHandler,
+        > = HashMap::new();
+
+        for h in resp.handlers.iter() {
+            let lock_option =
+                avalanche_types::subnet::rpc::common::http_handler::LockOptions::try_from(
```

Comment:
One thing I noticed is that we have really long import names. It might be helpful to think about using an alias for some imports. We could potentially address that in a follow-up.

Reply:
yeah this is a client only for the use of testing. I left the naming long on purpose so its easy to see the namespace but we could def nix avalanche_types prefix.